tests/rules/test_git_push.py

```diff
@@ -4,38 +4,51 @@
 
 
 @pytest.fixture
-def output():
-    return '''fatal: The current branch master has no upstream branch.
+def output(branch):
+    return '''fatal: The current branch {} has no upstream branch.
 To push the current branch and set the remote as upstream, use
 
-    git push --set-upstream origin master
-
-'''
-
-
-def test_match(output):
-    assert match(Command('git push', output))
-    assert match(Command('git push master', output))
-    assert not match(Command('git push master', ''))
-    assert not match(Command('ls', output))
-
-
-def test_get_new_command(output):
-    assert get_new_command(Command('git push', output))\
-        == "git push --set-upstream origin master"
-    assert get_new_command(Command('git push master', output))\
-        == "git push --set-upstream origin master"
-    assert get_new_command(Command('git push -u', output))\
-        == "git push --set-upstream origin master"
-    assert get_new_command(Command('git push -u origin', output))\
-        == "git push --set-upstream origin master"
-    assert get_new_command(Command('git push origin', output))\
-        == "git push --set-upstream origin master"
-    assert get_new_command(Command('git push --set-upstream origin', output))\
-        == "git push --set-upstream origin master"
-    assert get_new_command(Command('git push --quiet', output))\
-        == "git push --set-upstream origin master --quiet"
-    assert get_new_command(Command('git push --quiet origin', output))\
-        == "git push --set-upstream origin master --quiet"
-    assert get_new_command(Command('git -c test=test push --quiet origin', output))\
-        == "git -c test=test push --set-upstream origin master --quiet"
+    git push --set-upstream origin {}
+
+'''.format(branch, branch)
+
+
+@pytest.mark.parametrize('command', [
+    Command('git push', output('master')),
+    Command('git push master', output('master'))])
```

Comment:
This way `output` is not being used as a fixture. I'll comment on that ASAP.

Reply:
It's identical in use to `test_git_checkout.py`.

Reply:
Please consider the following:

```python
@pytest.fixture
def branch():
    return 'master'


# output fixture definition stays the same


@pytest.mark.parametrize('script', [
    'git push',
    'git push master'])
def test_match(script, output):
    assert match(Command(script, output))


@pytest.mark.parametrize('script, branch', [
    ('git push', 'it\'s-just-a-branch'),
    ('git push let\'s-push-it', 'let\'s-push-it')])
def test_match_branch(script, branch, output):
    assert match(Command(script, output))


# the rest must be adapted
```

What do you guys think of this change?

Reply:
Sure, I can give that a go.

Reply:
It's not obvious to me how a) your branch test fixture is used and b) what you expect the branch in test_match_branch to be used for?

Reply:
No worries, I think I've managed to sort it.  What do you think?